code/Path_Distance_TableGenerator.py

```diff
@@ -0,0 +1,311 @@
+        edge = ((fc, fs, fcc), (tc, ts, tcc))
+        global_edges_dict[edge] = {}
+        global_edges_dict[edge]['DIST_KM'] = dist_km
+        global_edges_dict[edge]['GEOM'] = wkt.loads(path_wkt)
+
+def create_graph():
+    # add nodes
+    for n in global_nodes_dict.keys():
+        x = global_nodes_dict[n]['x']
+        y = global_nodes_dict[n]['y']
+        geom = global_nodes_dict[n]['GEOM']
+        global_graph.add_node(n, x=x, y=y, geom=geom)
+    # add edges
+    for e in global_edges_dict.keys():
+        dist = global_edges_dict[e]['DIST_KM']
+        geom = global_edges_dict[e]['GEOM']
+        global_graph.add_edge(e[0], e[1], dist_km=dist, geom=geom)
+
+def initialize_the_global_graph(db_file):
+    global global_querier
+    global_querier = qdb.queryDatabase(db_file)
+    query_db_for_nodes()
+    query_db_for_edges()
+    create_graph()
+
+def get_shortest_path(src, dst, route_geom, waypoints_geom):
+    fc = src.split(',')[0].strip()
+    fs = src.split(',')[1].strip()
+    fcc = src.split(',')[2].strip()
+    tc = dst.split(',')[0].strip()
+    ts = dst.split(',')[1].strip()
+    tcc = dst.split(',')[2].strip()
+
+    src = (fc, fs, fcc)
+    dst = (tc, ts, tcc)
+
+    try:
+        route = nx.shortest_path(global_graph, src, dst, weight='dist_km')
+        for i in range(len(route)-1):
+            e = (route[i], route[i+1])
+            if not e in global_edges_dict.keys():
+                e = (route[i+1], route[i])
+            geom = global_edges_dict[e]["GEOM"]
+            route_geom.append(geom)
+            if not global_nodes_dict[e[0]]["GEOM"] in waypoints_geom:
+                waypoints_geom.append(global_nodes_dict[e[0]]["GEOM"])
+            if not global_nodes_dict[e[1]]["GEOM"] in waypoints_geom:
+                waypoints_geom.append(global_nodes_dict[e[1]]["GEOM"])
+        return route
+    except:
+        print(f"Could not complete query from '{src}' to '{dst}'.", file=sys.stderr)
+        return None
+
+def calc_dist_along_path(route):
+    dist = 0.0
+    for i in range(len(route)-1):
+        e = (route[i], route[i+1])
+        if not e in global_edges_dict.keys():
+            e = (route[i+1], route[i])
+        dist += global_edges_dict[e]["DIST_KM"]
+    return dist
+
+def transform_location_format(location):
+    return location.replace('/', ',')
+
+def find_shortest_path(db_file, start_city, start_state, start_country, end_city, end_state, end_country):
+    # Connect to the SQLite database
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+
+    # cursor.execute('UPDATE phys_nodes AS pnc \
+    # SET state = cp.state_province \
+    # FROM city_points AS cp \
+    # WHERE ABS(pnc.latitude - cp.city_latitude) < 0.001\
+    # AND ABS(pnc.longitude - cp.city_longitude) < 0.001 \
+    # AND pnc.city = cp.city_name \
+    # AND pnc.country = cp.country_code;')
+
+    # conn.commit()
+    # conn.close()
+
+    # Query the nodes and connections from the database
```

Comment:
Can we break down this long function into parts? E.g.
```
# fill_in_missing_states_in_phys_nodes()
g = create_graph_from_phys_nodes()
shortest_paths = get_shortest_paths(g, start_nodes, end_nodes)
shortest_path_distribution = get_shortest_path_distribution(shortest_paths)
print_shortest_paths_and_distances(shortest_path_distribution)
```

Reply:
done, now the function is divided into three subsection
create_graph_from_phys_nodes
shortest_path_distribution
print_shortest_paths_and_distances